frontend/src/components/pages/CourseDashboard.js

```diff
@@ -139,21 +139,41 @@ export function CourseDashboard() {
     );
 
     const myCourses = courseInfo.map((course) => (
-        <a href={`/courses/${course.ID}`} key={course.ID}>
-            <div className="bg-gray-100 p-4 rounded shadow hover:bg-gray-300">
-                <h3 className="text-xl font-semibold truncate overflow-hidden">{course.title}</h3>
-                <p className="mt-2">{course.description}</p>
+        <a href={`/courses/${course.ID}`} key={course.ID} className="max-w-[40vw] aspect-[9/8]">
+            <div className="bg-gray-100 rounded shadow hover:bg-gray-300 flex flex-col h-full w-full outline outline-1 outline-black/25">
+                <div className="h-[60%] ">
+                    <img
+                        className="rounded outline outline-1 outline-black/40 object-fill w-full h-full block aspect-[16/9]" 
+                        src={process.env.PUBLIC_URL + `/content/${course.ID}/thumbnail.png`} 
+                        onError={(e)=>{e.target.onError = null; e.target.src = `/default_thumbnails/tn${course.ID % 5}.png`}}
+                        alt="Thumbnail"
+                    />
+                </div>
+                <div className="h-[40%] p-2 md:p-4 xl:p-2 overflow-hidden">
+                    <h3 className="text-base font-semibold truncate overflow-hidden">{course.title}</h3>
+                    <p className="truncate overflow-hidden">{course.description}</p>
+                </div>
             </div>
         </a>
     ));
 
     const allCourses = filteredCourses.map((course) => (
-        <div key={course.ID} className="course-box">
-            <a href={`/courses/${course.ID}`} key={course.ID}>
-                <h3><b>{course.title}</b></h3>
-                <p>{course.description}</p>
-            </a>
-        </div>
+        <a href={`/courses/${course.ID}`} key={course.ID} className="max-w-[40vw] aspect-[9/8]">
+            <div className="bg-gray-100 rounded shadow hover:bg-gray-300 flex flex-col h-full w-full outline outline-1 outline-black/25">
+                <div className="h-[60%] ">
+                    <img
+                        className="rounded outline outline-1 outline-black/40 object-fill w-full h-full block aspect-[16/9]" 
+                        src={process.env.PUBLIC_URL + `/content/${course.ID}/thumbnail.png`} 
+                        onError={(e)=>{e.target.onError = null; e.target.className = "invisible"}}
```

Comment:
nit: Probably would want to have the same random thumbnails if none were added here:
```js
onError={(e)=>{e.target.onError = null; e.target.src = `/default_thumbnails/tn${course.ID % 5}.png`}}
```

This would prevent it to be transparent like here:
<img width="778" alt="Screenshot 2024-11-28 at 13 23 06" src="https://github.com/user-attachments/assets/27a9922c-db9c-419f-910f-162983ee1393">